File: manual/code/test_suites/cpp/FunctionBuildTest/src/function_datetime_strptime.cpp
#include <string>
#include <ctime>
#include <iomanip>
#include <sstream>

class DateParser {
public:
    std::string parse_date(const std::string& date_str, const std::string& format_str) {
        struct tm tm = {};
        std::istringstream ss(date_str);
        ss >> std::get_time(&tm, format_str.c_str());
        if (ss.fail()) {
            return "failed";
        }
        char buffer[20];
        strftime(buffer, sizeof(buffer), "%Y-%m-%d %H:%M:%S", &tm);
        return std::string(buffer);
    }
};

File: manual/data/python/src/function_datetime_strptime.py
from datetime import datetime

class DateParser:
    def parse_date(self, date_str: str, format_str: str) -> str:
        try:
            parsed_date = datetime.strptime(date_str, format_str)
            return parsed_date.strftime('%Y-%m-%d %H:%M:%S')
        except ValueError:
            return "failed"